application/src/main/java/org/togetherjava/tjbot/commands/moderation/modmail/ModMailCommand.java

```diff
@@ -0,0 +1,165 @@
+ * even if not part of the server anymore, e.g. after a ban.
+ * <p>
+ * Therefore, the user DMs the bot and the message are forwarded to moderators in a dedicated
+ * channel.
+ * <p>
+ * Actions are then logged to a dedicated channel, given by
+ * {@link Config#getModAuditLogChannelPattern()}.
+ */
+
+public final class ModMailCommand extends SlashCommandAdapter {
+
+    private static final Logger logger = LoggerFactory.getLogger(ModMailCommand.class);
+    private static final String COMMAND_NAME = "modmail";
+    private static final String OPTION_MESSAGE = "message";
+    private static final String OPTION_STAY_ANONYMOUS = "stay-anonymous";
+    private static final String OPTION_SERVER_GUILD = "server-guild";
+    private static final int COOLDOWN_DURATION_VALUE = 30;
+    private static final ChronoUnit COOLDOWN_DURATION_UNIT = ChronoUnit.MINUTES;
+    private static final Color AMBIENT_COLOR = Color.black;
+    private final Cache<Long, Instant> authorIdToLastCommandInvocation = createCooldownCache();
+    private final Predicate<String> modMailChannelNamePredicate;
+
+
+    /**
+     * Creates a new instance.
+     *
+     * @param jda the JDA instance to use to retrieve guildCache
+     * @param config the config to use for this
+     */
+    public ModMailCommand(JDA jda, Config config) {
+        super(COMMAND_NAME, "Send a message to the moderators of the selected guild",
+                CommandVisibility.GLOBAL);
+
+        OptionData messageOption = new OptionData(OptionType.STRING, OPTION_MESSAGE,
+                "What do you want to tell them?", true);
+        OptionData guildOption = new OptionData(OptionType.STRING, OPTION_SERVER_GUILD,
+                "Your server guild ID", true);
+        OptionData anonymousOption = new OptionData(OptionType.BOOLEAN, OPTION_STAY_ANONYMOUS,
+                "if set, your name is hidden", true);
+
+        List<Command.Choice> choices = jda.getGuildCache()
+            .stream()
+            .map(guild -> new Command.Choice(guild.getName(), guild.getIdLong()))
+            .toList();
+
+        guildOption.addChoices(choices);
+
+        getData().addOptions(messageOption, guildOption, anonymousOption);
+
+        modMailChannelNamePredicate =
+                Pattern.compile(config.getModMailChannelPattern()).asMatchPredicate();
+    }
+
+    private Cache<Long, Instant> createCooldownCache() {
+        return Caffeine.newBuilder()
+            .maximumSize(1_000)
+            .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+            .build();
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        long userId = event.getUser().getIdLong();
+        event.deferReply().setEphemeral(true).queue();
+
+        if (isChannelOnCooldown(userId)) {
+            event.reply("Can only be used once per %s minutes.".formatted(COOLDOWN_DURATION_VALUE))
+                .setEphemeral(true)
+                .queue();
```

Comment:
u cant use `event.reply(...)` after `event.deferReply()` anymore.

move the `defer` part below, when you are sure that u have to do long operations.